SAM2_Tracking/utils.py

```diff
@@ -35,6 +36,215 @@ def read_config_yaml(config_path):
+
+p
+
+    Special handling is included for the "video_frame_size" key, which may contain a list of
+    lists (e.g., [[1920, 1080], [1280, 720], ...]) and is included in the validation if so.
+
+    Parameters
+    ----------
+    configs : dict
+        Dictionary of configuration parameters, where each value is either a single value
+        or a list of values representing multiple trials.
+
+    Returns
+    -------
+    int
+        The number of trials specified across multi-trial configurations.
+
+    Raises
+    ------
+    ValueError
+        If multiple configuration keys have differing numbers of trials.
+    
+    Examples
+    --------
+    >>> configs = {
+    ...     'frame_dir': ['path1', 'path2'],
+    ...     'model_cfg': ['cfg1', 'cfg2'],
+    ...     'fps': 30
+    ... }
+    >>> extract_config_lens(configs)
+    There are 2 trials provided for processing.
+    2
+    """
+    # Get length of provided values for each listed config key as a dictionary 
+    config_counts = {key: len(value) for key, value in configs.items() if isinstance(value,list) 
+                     and key != "video_frame_size"}
+    # If multiple video_frame_size trials are provided, add count to config_counts
+    if lol_check(configs["video_frame_size"]):
+        config_counts["video_frame_size"] = len(configs["video_frame_size"])
+
+    # Extract the unique lengths of configuration values
+    unique_counts = set(config_counts.values())
+    
+    # Check for mismatches in provided configuration lengths,
+    # accounting for instances where the user provided a single value in list format
+    mismatch = len([x for x in unique_counts if x!= 1]) > 1
+
+ # Confirm that all provided configurations are in a list of the same length
+    if mismatch:
+        # Find the inconsistent key lists
+        # Raise a helpful error stating the length of each value for keys that have multiple trials
+        err_msg = "Inconsistent configuration lengths found:\n"
+        err_msg += "All configuration parameters that have \n"
+        err_msg += "multiple trials, must have the same length. \n"
+        for key, count in config_counts.items():
+            err_msg += f" - {key} trial count: {count}\n"
+        raise ValueError(err_msg)
+    else:
+        if len(unique_counts) > 1: # Multiple trials, some contain lists of a single entry
+            unique_counts.discard(1)
+            trial_count = unique_counts.pop()
+            print(f"There are {trial_count} trials provided for processing")
+            return trial_count
+        if not unique_counts: # The set is empty because all values are provided as a single entry
+            trial_count = 1
+            print(f" There is 1 trial provided for processing")
+            return trial_count
+        else: # All list configs are provided in the same length (no single lists)
+            trial_count = unique_counts.pop()
+            print(f"There are {trial_count} trials provided for processing")
```

Comment:
I suggest that we remove all of these print statements. This is great for debugging, but is not great for reusability and a print statement in each condition is not needed. For example, let's say for some reason we want to reuse this function for a completely different purpose, this output may not fit that purpose. Any printing of this information should happen after this function is called using the output of this function.

Reply:
See my comment in `run_segmentation` where you use this function.